### PR TITLE
Feature/add plot

### PR DESCRIPTION
**Pull request by tgurbich**
Added Circos plot generation for the merged GFF.
Added a retry label for UniFIRE.

### CODE REVIEW

File: bin/circos_plot.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        handles = handles + [
+            Patch(color="blue", label="Mobilome (phage)"),
+            Patch(color="lightseagreen", label="Mobilome (other)"),
+        ]
+
+    main_legend = circos.ax.legend(
+        handles=handles, bbox_to_anchor=(0.5, 0.475), loc="center", fontsize=8
+    )
+    circos.ax.add_artist(main_legend)
+    fig.savefig(outfile, dpi=600)
```
Comment (mberacochea) [inline bin/circos_plot.py:149]: I would make the dpi configurable, as an optional arg. But not a blocker for this PR

File: bin/circos_plot.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+
+def remove_escaped_characters(infile):
+    outfile = infile + "_modified"
+    with open(infile, "r") as file_in:
+        content = file_in.read()
+        modified_content = content.replace("\\=", "")
+
+    # Open the file in write mode to overwrite it with the modified content
+    with open(outfile, "w") as file_out:
```
Comment (mberacochea) [inline bin/circos_plot.py:159]: I would make a copy, re-writing files is not what most users would expect

File: modules/local/circos_plot.nf [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    input:
+    tuple val(meta), path(gff)
+
+    output:
+    tuple val(meta), path("*.png"), emit: circos_plot, optional: true
+
+    script:
+    """
+    circos_plot.py -p ${meta.prefix} -i ${gff} -o ${meta.prefix}_plot.png
```
Comment (mberacochea) [inline modules/local/circos_plot.nf:16]: It's missing the VERSION thingy
```
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pyCirclize: \$(python -c "import pycirclize; print(pycirclize.__version__)")
    END_VERSIONS
```

File: workflows/mettannotator.nf [hunk 4]
```diff
@@ -369,6 +370,10 @@ workflow METTANNOTATOR {
 
     ch_versions = ch_versions.mix(ANNOTATE_GFF.out.versions.first())
 
+    CIRCOS_PLOT (
+        ANNOTATE_GFF.out.annotated_gff
+    )
+
```
Comment (mberacochea) [inline workflows/mettannotator.nf:376]:
```suggestion
    ch_versions = ch_versions.mix(CIRCOS_PLOT.out.versions.first())
```

**Review by mberacochea — COMMENTED** [review]
_neat_ Thanks Tanya :). The code looks good to me. I find the circos plot extremely user friendly ( especially for those non-experienced users ).

I left 2 comments, related to the new module software version (the suggestions I made should work as they are but I didn't test them)

File: bin/circos_plot.py (hunk 2 above)
Comment (tgurbich) [reply bin/circos_plot.py:159]: It's a poorly worded comment on my part, it is making a copy. I will edit it for clarity.

File: bin/circos_plot.py (hunk 1 above)
Comment (tgurbich) [reply bin/circos_plot.py]: Good idea, added.

File: bin/circos_plot.py (hunk 2 above)
Comment (mberacochea) [reply bin/circos_plot.py:159]: ahh.. my bad. I misread the code

File: modules/local/circos_plot.nf (hunk 3 above)
Comment (tgurbich) [reply modules/local/circos_plot.nf:17]: thanks! done.

**Pull request merged**